### ISSUE TITLE
Preprocessing adjacency matrix 

### ISSUE THREAD

**Issue by leowyy** [tkipf/gcn#76]
Hi Thomas,

I'm slightly puzzled by a portion of code used for data preprocessing. In `utils.load_data()`, I noticed that the indices of features and labels are reordered according to the pre-saved test indices. However, the indices of the adjacency matrix are not reordered. Is this because the indices of the adjacency matrix were already reordered in your saved files?
Lines 69-73
```
    features[test_idx_reorder, :] = features[test_idx_range, :]
    adj = nx.adjacency_matrix(nx.from_dict_of_lists(graph))

    ...
    labels[test_idx_reorder, :] = labels[test_idx_range, :]
```
Thank you!

**Comment by tkipf** [comment]
Good catch. This might indeed be a bug, but given that this data loader has been used as a de facto benchmark now in quite a number of works (and also by predecessor work by Yang et al., ICML 2016), it is best to just keep it as is if you’d like to compare numbers.

If you’re interested in a more instructive/flexible data loader for these citation network datasets, have a look at: https://github.com/tkipf/pygcn <https://github.com/tkipf/pygcn>. Note that the dataset splits are different here, but this allows you to choose your own splits/run your own evaluation protocols more easily (e.g. test on different random splits etc.). In general I’d recommend against using the fixed split from Yang et al.’s paper (which we also use in this repo) these days, as it is very easy to overfit on the validation set, which affects many recent works.

**Comment by signapoop** [comment]
Thanks for your response; I'll take a look at the data loader in the repo you've mentioned.

Merry Christmas!

**Comment by DavideBuffelli** [comment]
I know this issue is closed but I think it could be very useful to present a fix for this bug, specially for future research. I think this should work:
```
....
features[test_idx_reorder, :] = features[test_idx_range, :]
G = nx.from_dict_of_lists(graph)
G = nx.relabel_nodes(G, {i: j for i,j in zip(test_idx_reorder, test_idx_range)})
adj = nx.adjacency_matrix(G)
...
```
